src/main/java/com/epam/aidial/core/controller/ShareController.java

```diff
@@ -137,6 +147,49 @@ public Future<?> revokeSharedResources() {
                 .onFailure(this::handleServiceError);
     }
 
+    public Future<?> copySharedAccess() {
+        return context.getRequest()
+                .body()
+                .compose(buffer -> {
+                    CopySharedAccessRequest request;
+                    try {
+                        request = ProxyUtil.convertToObject(buffer, CopySharedAccessRequest.class);
+                    } catch (Exception e) {
+                        log.error("Invalid request body provided", e);
+                        throw new IllegalArgumentException("Can't initiate copy shared access request. Incorrect body provided");
+                    }
+
+                    String sourceUrl = request.sourceUrl();
```

Comment:
It could be easier to set constraint on JSON fields instead of writing checks in the controller

Reply:
I prefer a user friendly error messages in this case (only 2 checks compare to conversation validation)

Reply:
ok. let's keep as is